src/ecs/util.ts

```diff
@@ -71,14 +71,28 @@ export async function checkPermissionsForSsm(
 export async function prepareCommand(
     client: EcsClient,
     command: string,
+    taskRoleArn: string,
     task: EcsTaskIdentifer
 ): Promise<{ path: string; args: string[]; dispose: () => void }> {
     const ssmPlugin = await getOrInstallCli('session-manager-plugin', !isCloud9())
-    const { session } = await client.executeCommand({ ...task, command })
+
+    let session: Session
+    try {
+        session = (await client.executeCommand({ ...task, command })).session!
+    } catch (execErr) {
+        await checkPermissionsForSsm(new DefaultIamClient(globals.regionProvider.defaultRegionId), {
+            taskRoleArn: taskRoleArn,
+        }).catch(permErr => {
+            throw ToolkitError.chain(permErr, `${execErr}`)
```

Comment:
Throwing ToolkitError is generally a good idea, so thanks for noticing that.

One thing to note: stringifying errors generally isn't useful (the stacktrace is useless in a non-debug build). Look at existing usages of `ToolkitError.chain` to see the recommended pattern. The "chained" error message can give a more semantic explanation since we are handling it in a place that has more context

    throw ToolkitError.chain(permErr, 'ECS exec failed')

, or if not then perhaps print the `.message`. The stacktrace will just be noise (and this is surfaced in customer-facing messages by the [global handler](https://github.com/justinmk3/aws-toolkit-vscode/blob/106666e0bbac3236b7ef50f45dd81634d6441d05/src/extension.ts#L95-L98))

    throw ToolkitError.chain(permErr, `${(execErr as Error).message}`)